bullet: fix style and some audit checks

scpeters opened the pull request:
Replaced all single quotes with double quotes
Removed require formula
Use sha256
"make install" -> "make", "install"
Use || instead of or

FYI: this project is now on github, but I'm waiting on the upstream developer to create tags for previous google code releases (https://github.com/bulletphysics/bullet3/issues/339) so we can stop pointing to google code. At that time, I was planning to point `head` to github as well.

There are still a few complaints from `brew audit --strict`, but I was going to save them for another day.
~~~
bullet:
 * A `test do` test block should be added
 * Options should begin with with/without. Migrate '--framework' with `deprecated_option`.
 * Options should begin with with/without. Migrate '--shared' with `deprecated_option`.
 * Options should begin with with/without. Migrate '--build-demo' with `deprecated_option`.
 * Options should begin with with/without. Migrate '--build-extra' with `deprecated_option`.
 * Options should begin with with/without. Migrate '--double-precision' with `deprecated_option`.
~~~

mikemcquaid commented:
> There are still a few complaints from brew audit --strict, but I was going to save them for another day.

Would be great to do them in this PR, if that's OK.

scpeters commented:
I guess it's after midnight, so that counts as another day. I fixed the rest of the audit checks and updated the HEAD as well to point to the current github repository.

scpeters commented:
This is a separate issue, but it looks like formulae that depend on `bullet` with the deprecated options don't recognize them, either with `brew audit` or `brew install`.
~~~
brew audit osrf/simulation/gazebo6
gazebo6:
 * Dependency bullet does not define option "shared"
 * Dependency bullet does not define option "double-precision"
 * Dependency bullet does not define option "shared"
 * Dependency bullet does not define option "double-precision"

Error: 4 problems in 1 formulae
~~~
I would expect a deprecation warning from `brew audit`, but it doesn't seem to see the old options at all.

mikemcquaid commented:
I believe it should pass through the options anyway (you might want to test that though) but if it does not the best bet is probably just to queue up a PR there to use the new names and get it merged shortly after this one.

scpeters commented:
I did test it and it didn't work, though I didn't paste those results. I'll see if I can make a trivial example.

scpeters commented:
I created #38069 to address the deprecated option issue. As you suggest, for now I'll just update my formulae that use bullet shortly after this is merged.

I'll squash when things are looking ok.

scpeters commented:
I just squashed, so this should be ready for merging unless someone finds any more problems.

scpeters commented:
Sorry I thought things were good to go, but something failed after I squashed, and I forgot to check back until now. (Are there email notifications for build failures?) I'm going to close and re-open to see if that triggers a new build, since details are gone for the old one.

scpeters commented:
Ok, now it passes CI. Ready for final review.

mikemcquaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

scpeters commented:
You're too kind. :smile: Thanks for all the work you do as a maintainer!
